### PR TITLE
verifyAttributePresent() method

### PR DESCRIPTION
**Pull request by egecansen**
- added method -> new attributePresent() & verifyAttributePresent()
- improved the bundleInteraction() method
- Fixed the unit tests

### CODE REVIEW

File: src/main/java/pickleib/web/utilities/WebUtilities.java [hunk 1]
```diff
@@ -357,4 +354,47 @@ public void fillIframeForm(
...
+    /**
+     *
+     * Verify that the attribute {attribute name} present for the {element name}
+     *
+     * @param element target element
+     * @param attributeName target attribute name
+     */
+    public boolean attributeNamePresent(
+            String attributeName,
+            WebElement element) {
```
Comment (Umutayb) [inline src/main/java/pickleib/web/utilities/WebUtilities.java:368]: Remove unnecessary lines

File: src/main/java/pickleib/web/utilities/WebUtilities.java [hunk 2]
```diff
@@ -357,4 +354,47 @@ public void fillIframeForm(
...
+
+
+        long initialTime = System.currentTimeMillis();
+        String caughtException = null;
+        int counter = 0;
+        do {
+            try {
+                Set<String> elementAttributes = getElementJson(element).getAsJsonObject().asMap().keySet();
+                return elementAttributes.contains(attributeName);
```
Comment (Umutayb) [inline src/main/java/pickleib/web/utilities/WebUtilities.java:377]: This does not wait until the attribute is contained, please fix

File: src/main/java/pickleib/web/utilities/WebUtilities.java [hunk 3]
```diff
@@ -357,4 +354,47 @@ public void fillIframeForm(
...
+                    log.warning("Iterating... (" + webDriverException.getClass().getName() + ")");
+                    caughtException = webDriverException.getClass().getName();
+                }
+                waitFor(0.5);
+                counter++;
+            }
+        }
+        while (!(System.currentTimeMillis() - initialTime > elementTimeout));
+        if (counter > 0) log.warning("Iterated " + counter + " time(s)!");
+        log.warning("Element does not contain " +
```
Comment (Umutayb) [inline src/main/java/pickleib/web/utilities/WebUtilities.java:394]: Unnecessary break

**Review by Umutayb — CHANGES_REQUESTED** [review]

File: src/main/java/pickleib/web/utilities/WebUtilities.java (hunk 1 above)
Comment (egecansen) [reply src/main/java/pickleib/web/utilities/WebUtilities.java]: done

File: src/main/java/pickleib/web/utilities/WebUtilities.java (hunk 3 above)
Comment (egecansen) [reply src/main/java/pickleib/web/utilities/WebUtilities.java]: done

File: src/main/java/pickleib/web/utilities/WebUtilities.java [hunk 4]
```diff
@@ -357,4 +354,41 @@ public void fillIframeForm(
         }
         driver.switchTo().parentFrame();
     }
+
```
Comment (Umutayb) [inline src/main/java/pickleib/web/utilities/WebUtilities.java:357]: Space remains...

File: src/main/java/pickleib/web/utilities/WebUtilities.java [hunk 5]
```diff
@@ -357,4 +354,41 @@ public void fillIframeForm(
...
+     * @param attributeName target attribute name
+     */
+    public boolean attributeNamePresent(String attributeName, WebElement element) {
+        long initialTime = System.currentTimeMillis();
+        String caughtException = null;
+        int counter = 0;
+        do {
+            try {
+                Set<String> elementAttributes = getElementJson(element).getAsJsonObject().asMap().keySet();
+                for (String attribute : elementAttributes) return attribute.equals(attributeName);
```
Comment (Umutayb) [inline src/main/java/pickleib/web/utilities/WebUtilities.java:373]: The method name suggests that the method is supposed to check if an attribute is present, however the returned value is whether this attribute ia equal to certain other value.. 

Please resolve this discrepancy.

**Review by Umutayb — CHANGES_REQUESTED** [review]

**Comment by egecansen** [comment]
can you check the last updates?

**Pull request closed without merging**